### PR TITLE
minidlna, bump old bep file

### PR DESCRIPTION
**Pull request by Begasus**
Build still fails due to some "expected specifier-qualifier-list", wip, so if someone could find a fix ... ;)

### CODE REVIEW

File: net-misc/minidlna/minidlna-1.2.1.recipe [hunk 1]
```diff
@@ -0,0 +1,80 @@
+SUMMARY="A server software compliant with multiple clients"
+DESCRIPTION="MiniDLNA MiniDLNA (aka ReadyDLNA) is server software with the \
+aim of being fully compliant with DLNA/UPnP-AV clients."
+HOMEPAGE="http://sourceforge.net/projects/minidlna/"
+COPYRIGHT="2008-2009  Justin Maggard
+	2008-2009  Justin Maggard (portions of MiniUPnP project)"REVISION="1"
+LICENSE="GNU GPL v2
+	BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="http://sourceforge.net/projects/minidlna/files/minidlna/$portVersion/minidlna-$portVersion.tar.gz"
```
Comment (fbrosson) [inline net-misc/minidlna/minidlna-1.2.1.recipe]:
```diff
-SOURCE_URI="http://sourceforge.net/projects/minidlna/files/minidlna/$portVersion/minidlna-$portVersion.tar.gz"
+SOURCE_URI="https://downloads.sourceforge.net/minidlna//minidlna-$portVersion.tar.gz"
```

File: net-misc/minidlna/minidlna-1.2.1.recipe [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+BUILD()
+{
+	autoreconf -vfi
+	LDFLAGS="-lnetwork" runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	#cd minidlna-1.0.21
```
Comment (fbrosson) [inline net-misc/minidlna/minidlna-1.2.1.recipe]:
```diff
-	#cd minidlna-1.0.21
```

Comment (Begasus) [reply net-misc/minidlna/minidlna-1.2.1.recipe]: Forgot to remove that one ;)

File: net-misc/minidlna/minidlna-1.2.1.recipe [hunk 3]
```diff
@@ -0,0 +1,79 @@
+SUMMARY="A server software compliant with multiple clients"
+DESCRIPTION="MiniDLNA MiniDLNA (aka ReadyDLNA) is server software with the \
+aim of being fully compliant with DLNA/UPnP-AV clients."
+HOMEPAGE="http://sourceforge.net/projects/minidlna/"
```
Comment (fbrosson) [inline net-misc/minidlna/minidlna-1.2.1.recipe]: Oops, I forgot to suggest something for `HOMEPAGE`:
* _a)_
```diff
-HOMEPAGE="https://sourceforge.net/projects/minidlna/"
+HOMEPAGE="http://minidlna.sourceforge.net/
+	https://sourceforge.net/projects/minidlna/"
```

Other suggestions:

* _b)_
```
HOMEPAGE="https://sourceforge.net/projects/minidlna/"
```

* _c)_
```
HOMEPAGE="http://minidlna.sourceforge.net/"
```
(I don't know which is best. _b_ has a link to _c_, and _c_ has a link to _b_ :)

**Comment by alaviss** [comment]
@Begasus 
> Part of the errors I got from building minidlna: https://pastebin.com/ajWB7YVz
Not sure if it's worth investigating and fixing them ...

Try adding `-D_BSD_SOURCE` to `CPPFLAGS` and `-lbsd` to `LIBS`. That should solve most, if not all errors :)

**Comment by Begasus** [comment]
@alaviss now I'm getting error: 'struct dirent' has no member named 'd_type' (investigating) ...

File: net-misc/minidlna/minidlna-1.2.1.recipe [hunk 4]
```diff
@@ -0,0 +1,83 @@
+SUMMARY="A server software compliant with multiple clients"
+DESCRIPTION="MiniDLNA MiniDLNA (aka ReadyDLNA) is server software with the \
+aim of being fully compliant with DLNA/UPnP-AV clients."
+HOMEPAGE="http://minidlna.sourceforge.net/
+	https://sourceforge.net/projects/minidlna/"
+COPYRIGHT="2008-2017  Justin Maggard"
+LICENSE="GNU GPL v2
+	BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/minidlna//minidlna-$portVersion.tar.gz"
```
Comment (alaviss) [inline net-misc/minidlna/minidlna-1.2.1.recipe]:
```diff
-SOURCE_URI="https://downloads.sourceforge.net/minidlna//minidlna-$portVersion.tar.gz"
+SOURCE_URI="https://downloads.sourceforge.net/minidlna/minidlna-$portVersion.tar.gz"

File: net-misc/minidlna/minidlna-1.2.1.recipe [hunk 5]
```diff
@@ -0,0 +1,83 @@
+SUMMARY="A server software compliant with multiple clients"
+DESCRIPTION="MiniDLNA MiniDLNA (aka ReadyDLNA) is server software with the \
+aim of being fully compliant with DLNA/UPnP-AV clients."
+HOMEPAGE="http://minidlna.sourceforge.net/
+	https://sourceforge.net/projects/minidlna/"
+COPYRIGHT="2008-2017  Justin Maggard"
```
Comment (alaviss) [inline net-misc/minidlna/minidlna-1.2.1.recipe]:
```diff
-COPYRIGHT="2008-2017  Justin Maggard"
+COPYRIGHT="2008-2017 Justin Maggard"

File: net-misc/minidlna/patches/minidlna-1.2.1.patchset [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
++
++		#ifndef __HAIKU__
+ 		switch( e->d_type )
+ 		{
+ 			case DT_DIR:
+@@ -543,6 +545,7 @@ monitor_insert_directory(int fd, char *name, const char * path)
+ 			default:
+ 				break;
+ 		}
++		#endif
```
Comment (alaviss) [inline net-misc/minidlna/patches/minidlna-1.2.1.patchset]: With your patch, minidlna might not correctly handle nested directories, hopefully this small change would work without any ill effects.

```diff
-+		#endif
++		#else
++		type = resolve_unknown_type(path_buf, dir_types);
++		#endif
```

File: net-misc/minidlna/patches/minidlna-1.2.1.patchset [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+ 
+ 	INIT_STR(str, header);
+ 
++#ifndef __HAIKU__
+ #if USE_FORK
+ 	if( (h->reqflags & FLAG_XFERBACKGROUND) && (setpriority(PRIO_PROCESS, 0, 19) == 0) )
+ 		tmode = "Background";
+ 	else
++	if( (h->reqflags & FLAG_XFERBACKGROUND)
++		tmode = "Background";
```
Comment (alaviss) [inline net-misc/minidlna/patches/minidlna-1.2.1.patchset]: I believe these were added by mistake :)

```diff
-+	if( (h->reqflags & FLAG_XFERBACKGROUND)
-+		tmode = "Background";

File: net-misc/minidlna/patches/minidlna-1.2.1.patchset [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
```
Comment (alaviss) [inline net-misc/minidlna/patches/minidlna-1.2.1.patchset]:
```diff
-+	if( (h->reqflags & FLAG_XFERBACKGROUND)
-+		tmode = "Background";

**Comment by Begasus** [comment]
Updated :)

**Comment by waddlesplash** [comment]
Does it work now?

**Pull request merged**